x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea } from 'kea';
+import { ReactNode } from 'react';
+import { History } from 'history';
+
+import { IKeaLogic, TKeaReducers, IKeaParams } from '../types';
+
+export interface IFlashMessage {
+  type: 'success' | 'info' | 'warning' | 'error';
+  message: ReactNode;
+  description?: ReactNode;
+}
+
+export interface IFlashMessagesLogicValues {
+  messages: IFlashMessage[];
+  queuedMessages: IFlashMessage[];
+  historyListener: Function | null;
+}
+export interface IFlashMessagesLogicActions {
+  setMessages(messages: IFlashMessage | IFlashMessage[]): void;
+  clearMessages(): void;
+  setQueuedMessages(messages: IFlashMessage | IFlashMessage[]): void;
+  clearQueuedMessages(): void;
+  listenToHistory(history: History): void;
+  setHistoryListener(historyListener: Function): void;
+}
+
+const convertToArray = (messages: IFlashMessage | IFlashMessage[]) =>
+  !Array.isArray(messages) ? [messages] : messages;
+
+export const FlashMessagesLogic = kea({
+  actions: (): IFlashMessagesLogicActions => ({
+    setMessages: (messages) => ({ messages: convertToArray(messages) }),
+    clearMessages: () => null,
+    setQueuedMessages: (messages) => ({ messages: convertToArray(messages) }),
+    clearQueuedMessages: () => null,
+    listenToHistory: (history) => history,
+    setHistoryListener: (historyListener) => ({ historyListener }),
+  }),
+  reducers: (): TKeaReducers<IFlashMessagesLogicValues, IFlashMessagesLogicActions> => ({
+    messages: [
```

Comment:
Also want to confidence check something by y'all - I simplified these names from
- `flashMessages`-> `messages` 
- `setFlashMessages`->`setMessages`

I figured it was an OK change to make since almost every file should be directly importing `FlashMessagesLogic` or calling `FlashMessagesLogic.actions.something` directly, making it fairly obvious what kind of messages these are.

Let me know if that's fine with everyone if you'd prefer I change it back to the more verbose naming.

Reply:
If these are typically imported and destructured from import statements in other logic files, then I think the more explicit naming would be better.

I don't know how we anticipate using these in, but if there's anywhere we'd be destructing from store values then more explicit would probably be better.

But we're not really using this anywhere yet so it's hard to tell. You could just leave it as is and we could change it later if it is a problem.

Reply:
If we destructured, it would happen immediately before the method was called, so for example:

```js
import { FlashMessagesLogic } from '../shared/flash_messages';

export const SomeOtherLogic = kea({
  // ...
  listeners: () => ({
    someListener: () => {
      const { setMessages } = FlashMessages.actions;
      // or
      FlashMessagesLogic.actions.setMessages('whatever');
    },
  }),
};
```

At least that's how we've done it so far in every App Search logic file 🤷

Reply:
I'd prefer the explicit naming. `messages` implies that there will never be any other concept of a "message" than a flash message and I think this removes any possibility of ambiguity

Reply:
👍 I'm good with that, will make the change!

Reply:
Resolved in ec50b60 - now using more explicit method name